### PR TITLE
updates to accept zip input format for Bulk endpoint

### PR DESCRIPTION
**Pull request by shans-qontigo**
Updates to accept zip input format for Bulk endpoint:
1. Updated bulk.py - to add additional headers
2. Updates session.py as currently is accepts only json input

### CODE REVIEW

**Review by lswift-qontigo — CHANGES_REQUESTED** [review]
On the face of it the changes look ok however, I don't think that compressed payloads are unique to the bulk api. I am wondering whether we would want to make the change in the actual session class instead of isolating the change to the bulk api.

File: axiomapy/axiomaapi/bulk.py [hunk 1]
```diff
@@ -30,21 +30,30 @@ class BulkAPI:
 
     @staticmethod
     def patch_portfolios_payload(
-        as_of_date: str, payload: dict, return_response: bool = True,
+        as_of_date: str,
+            payload,
```
Comment (wdeng-qontigo) [inline axiomapy/axiomaapi/bulk.py:34]: type for "payload" is missing

File: axiomapy/session.py [hunk 2]
```diff
@@ -508,6 +508,7 @@ def _prepare_request_args(
         method: HttpMethods,
         url: str,
         json: dict = None,
+        data = None,
```
Comment (wdeng-qontigo) [inline axiomapy/session.py:511]: type for "data"?

File: axiomapy/session.py [hunk 3]
```diff
@@ -767,7 +772,8 @@ def _put(
     def _patch(
         self,
         url: str,
-        json: dict,
+        json: dict = None,
+        data = None,
```
Comment (wdeng-qontigo) [inline axiomapy/session.py:776]: type of "data"?

File: axiomapy/axiomaapi/bulk.py (hunk 1 above)
Comment (shans-qontigo) [reply axiomapy/axiomaapi/bulk.py]: Added!

File: axiomapy/session.py (hunk 3 above)
Comment (shans-qontigo) [reply axiomapy/session.py]: Added

**Review by lswift-qontigo — APPROVED** [review]

**Review by bspector-qontigo — APPROVED** [review]

**Review by wdeng-qontigo — APPROVED** [review]

**Pull request merged**